### PR TITLE
fix issue#132

### LINKED ISSUE #132: Allow blockHash argument in eth_call
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1898.md should be supported (apparently currently it's not) 

```
curl -s -H Content-Type:application/json -X POST --data '{"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0x95d89b41","gas":"0x2faf080","to":"0x2ad8d8ac1e919a2402fc2d2f0dda4b23ddb25e8b"},"0xf66b4d8e26896d184f3a76986149b6483870ec98e7764bd03597ebee63269ae6"],"id":45}' https://testnet.emerald.oasis.dev
{"jsonrpc":"2.0","id":45,"error":{"code":-32602,"message":"invalid argument 1: hex number \u003e 64 bits"}}

```

### PR DESCRIPTION
**Pull request by lesterli**
Fix #132, please review and merge it, thanks.

### CODE REVIEW

File: rpc/eth/api.go [hunk 1]
```diff
@@ -673,3 +674,19 @@ func (api *PublicAPI) Hashrate() hexutil.Uint64 {
...
 	return 0
 }
+
+// getBlockRound returns the block round from BlockNumberOrHash.
+func (api *PublicAPI) getBlockRound(logger *logging.Logger, blockNrOrHash ethrpc.BlockNumberOrHash) (uint64, error) {
+	switch {
+	case blockNrOrHash.BlockHash == nil && blockNrOrHash.BlockNumber == nil:
+		return 0, fmt.Errorf("types BlockHash and BlockNumber cannot be both nil")
+	case blockNrOrHash.BlockHash != nil:
+		round, err := api.backend.QueryBlockRound(*blockNrOrHash.BlockHash)
```
Comment (kostko) [inline rpc/eth/api.go:684]:
```suggestion
		return api.backend.QueryBlockRound(*blockNrOrHash.BlockHash)
```

File: rpc/eth/api.go [hunk 2]
```diff
@@ -673,3 +674,19 @@ func (api *PublicAPI) Hashrate() hexutil.Uint64 {
...
+		return round, err
+	case blockNrOrHash.BlockNumber != nil:
+		round, err := api.roundParamFromBlockNum(logger, *blockNrOrHash.BlockNumber)
```
Comment (kostko) [inline rpc/eth/api.go:687]:
```suggestion
		return api.roundParamFromBlockNum(logger, *blockNrOrHash.BlockNumber)
```

**Review by kostko — COMMENTED** [review]
Please add some tests for the newly supported variants.

**Review by ptrus — CHANGES_REQUESTED** [review]
Can the following methods also be updated:
- `eth_getBalance`
- `eth_getStorageAt`
- `eth_getTransactionCount`
- `eth_getCode`

so that the full EIP-1898 is implemented (https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1898.md).

**Comment by ptrus** [comment]
This was done in https://github.com/oasisprotocol/emerald-web3-gateway/pull/173

**Pull request closed without merging**